parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -737,21 +758,17 @@ pub mod pallet {
 				slot
 			);
 
-			let mut last_finalized_header = <LatestFinalizedHeaderState<T>>::get();
-			let latest_finalized_header_slot = last_finalized_header.beacon_slot;
-
-			if slot > latest_finalized_header_slot {
```

Comment:
is the if logic nessessary here? should the finalized header update always continuously increasing?

Reply:
What if a relayer provides an older finalized update than a newer one that we've received? Then they would set the latest finalized header backwards, and it could lead to some weird state. For example:

Receive finalized header at slot 100
Validate message at slot 90
Relayer sends finalized header at slot 80

Now the beacon client sees the latest finalized header at slot 80 as the finalized header in the chain, but it has already validated messages that are children of the finalized block.

Do you have a scenario where it would be necessary to allow an older finalized block to be set as the latest finalized header?

Reply:
> Do you have a scenario where it would be necessary to allow an older finalized block to be set as the latest finalized header?
no, I would assume this won't happen so add check here, then the if check seems unnessessary to me
https://github.com/Snowfork/snowbridge/blob/de2fc20009850297b5bc560f894bc68d55e67baf/parachain/pallets/ethereum-beacon-client/src/lib.rs#L432

Reply:
I see! Cool, that makes it cleaner. ✨